### PR TITLE
add type definition previewer

### PR DESCRIPTION
**Pull request by Xrayya**
https://user-images.githubusercontent.com/90083127/179449442-9184f244-fd17-486b-ac6f-cbf7808b1635.mov

### CODE REVIEW

File: lua/goto-preview.lua [hunk 1]
```diff
@@ -120,6 +135,12 @@ M.apply_default_mappings = function()
       "<cmd>lua require('goto-preview').goto_preview_definition()<CR>",
       { noremap = true }
     )
+    vim.api.nvim_set_keymap(
+      "n",
+      "gpt",
+      "<cmd>lua require('goto-preview').goto_preview_type_implementation()<CR>",
```
Comment (rmagatti) [inline lua/goto-preview.lua:141]: You probably meant to have this instead?
```suggestion
      "<cmd>lua require('goto-preview').goto_preview_type_definition()<CR>",
```

**Review by rmagatti — CHANGES_REQUESTED** [review]

**Comment by rmagatti** [comment]
Thanks for the PR! Just a small thing I noticed on the keybindings.

File: lua/goto-preview.lua (hunk 1 above)
Comment (Xrayya) [reply lua/goto-preview.lua:141]: Ahh yes, I'm sorry, I wasn't aware of that

**Comment by Xrayya** [comment]
Pardon, can I ask, did I do another mistakes? Cuz this is my second PR since I register my github account and I still lack knowledge of many things

**Review by rmagatti — APPROVED** [review]
Looks good to me

**Comment by rmagatti** [comment]
@Xrayya, no this looks good to me, merging!

**Comment by Xrayya** [comment]
Okay, nice then

**Pull request merged**